stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2021 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+from odoo.tools import float_round
+
+
+class DeliverySlipReport(models.AbstractModel):
+    _name = "report.stock.report_deliveryslip"
+    _description = "Delivery Slip Report"
+
+    @api.model
+    def get_remaining_to_deliver_data(self, picking):
+        """Return dictionaries encoding pending quantities to deliver
+
+        Returns a list of dictionaries, encoding the data to be displayed
+        at the end of the delivery slip, summarising for each order the
+        pending quantities for each of its lines.
+        """
+        remaining_to_deliver_data = []
+        report_lines = picking.get_delivery_report_lines()
```

Comment:
hmm them this method is going to be called twice: 1st in the report, 2nd here. 
I'd compute this before reaching the report or from the report by passing here the lines.

Reply:
Done.